ostd/src/mm/vm_space.rs

```diff
@@ -291,6 +291,14 @@ impl CursorMut<'_> {
         Ok(self.0.query().map(|item| item.try_into().unwrap())?)
     }
 
+    /// Traverses forward in the current level to the next PTE.
+    ///
+    /// If reached the end of a page table node, it leads itself up to the next page of the parent
+    /// page if possible.
+    pub fn move_forward(&mut self) {
```

Comment:
This API is not really usable as how far it moves forward depends on the level. The user should use `jump` to jump to a certain virtual address instead.

Reply:
I am currently using `jump` for correction, but I find that `jump` seems somewhat less user-friendly in this traversal scenario. For instance, in the current code, I have to add an additional check to prevent `jump` from touching the boundary during the last iteration, which will trigger a panic. Is there a more elegant way to use `jump` at present? Or could the API behavior of `jump` when it goes beyond the boundary be changed?

Reply:
You can introduce a `jump_forward` method or let the `jump` method return a `Result` instead. At your convenience.